token-locker/src/lib.rs

```diff
@@ -274,8 +324,10 @@ impl Contract {
         #[serializer(borsh)] proof: Proof,
     ) -> Promise {
         assert!(verification_success, "Failed to verify the proof");
-        let required_deposit = self.record_proof(&proof);
 
+        // The deposit required to cover the cost
+        // of additional storage of used proof
+        let required_deposit = self.record_proof(&proof);
         assert!(env::attached_deposit() >= required_deposit);
```

Comment:
It also should be an extra 2 YOCTO which we used for ft_transfer_call. And maybe we want to return extra money back if the deposit is bigger than required.